src/core/public/http/abortable.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { HttpHandler, HttpFetchOptions, AbortablePromise } from './types';
+
+export function abortable<T>(handler: HttpHandler<T>) {
+  return (path: string, options: HttpFetchOptions = {}): AbortablePromise<T> => {
+    const controller = options.signal
+      ? new AbortController()
+      : { signal: options.signal, abort: Function.prototype };
+    const promise = new Promise<T>((resolve, reject) => {
+      handler(path, { ...options, signal: controller.signal }).then(resolve, reject);
+    });
+
+    // NOTE: We are only using Object.defineProperty with enumerable:false to be explicit about
+    // excluding promise.abort() which isn't the case with Object.assign.
+    return Object.defineProperty(promise, 'abort', {
+      enumerable: false,
```

Comment:
add `configurable: false`?

Reply:
The reason we are explicit about excluding `abort` from being enumerable is in case someone is testing for equality on a Promise instance would normally be externally iterable. With `configurable: false`, it isn't really changing how any testing is done, and is already the default, so I don't think provides value here.

Reply:
Ah I didn't know it was a default 👍